packages/skygear-react-native/ios/SGSkygearReactNative.m

```diff
@@ -0,0 +1,353 @@
+                    self.openURLResolve = nil;
+                    self.openURLReject = nil;
+                }
+            } else {
+                if (self.openURLResolve) {
+                    self.openURLResolve([url absoluteString]);
+                    self.openURLResolve = nil;
+                    self.openURLReject = nil;
+                }
+            }
+            self.sfSession = nil;
+        }];
+        [self.sfSession start];
+    } else if (@available(iOS 10.0, *)) {
+        [RCTSharedApplication() openURL:url options:@{} completionHandler:^(BOOL success) {
+            if (!success) {
+                if (self.openURLReject) {
+                    self.openURLReject(RCTErrorUnspecified, [NSString stringWithFormat:@"Unable to open URL: %@", url], nil);
+                    self.openURLResolve = nil;
+                    self.openURLReject = nil;
+                }
+            }
+        }];
+    } else {
+        BOOL opened = [RCTSharedApplication() openURL:url];
+        if (!opened) {
+            if (self.openURLReject) {
+                self.openURLReject(RCTErrorUnspecified, [NSString stringWithFormat:@"Unable to open URL: %@", url], nil);
+                self.openURLResolve = nil;
+                self.openURLReject = nil;
+            }
+        }
+    }
+}
+
+RCT_EXPORT_METHOD(signInWithApple:(NSString *)url
+                          resolve:(RCTPromiseResolveBlock)resolve
+                           reject:(RCTPromiseRejectBlock)reject)
+{
+  if (@available(iOS 13.0, *)) {
+    NSString *state = nil;
+    NSString *nonce = nil;
+    NSURLComponents *components = [NSURLComponents componentsWithString:url];
+    for (NSURLQueryItem *item in components.queryItems) {
+      if ([item.name isEqualToString:@"state"]) {
+        state = item.value;
+      }
+      if ([item.name isEqualToString:@"nonce"]) {
+        nonce = item.value;
+      }
+    }
+
+    if (!state || !nonce) {
+      reject(RCTErrorUnspecified, [NSString stringWithFormat:@"Invalid URL: %@", url], nil);
+      return;
+    }
+
+    ASAuthorizationAppleIDProvider *provider = [[ASAuthorizationAppleIDProvider alloc] init];
+
+    ASAuthorizationAppleIDRequest *request = [provider createRequest];
+    request.requestedOperation = ASAuthorizationOperationLogin;
+    request.requestedScopes = @[ASAuthorizationScopeEmail];
+    request.state = state;
+    request.nonce = nonce;
+
+    ASAuthorizationController *controller = [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[request]];
+    controller.presentationContextProvider = self;
+    controller.delegate = self;
+
+    self.provider = provider;
+    self.controller = controller;
+    self.state = state;
+    self.nonce = nonce;
+    self.signInWithAppleResolve = resolve;
+    self.signInWithAppleReject = reject;
+
+    [controller performRequests];
+  } else {
+    reject(RCTErrorUnspecified, @"Sign in with Apple requires iOS 13", nil);
```

Comment:
Do we want to support older version through standard OIDC flow/JS SDK?

Reply:
explained in a previous comment.

Reply:
If so, we may want to have a more descriptive error message, e.g. "use standard OIDC flow instead of this function".